core/primitives/src/errors.rs

```diff
@@ -83,8 +83,7 @@ pub enum StorageError {
     /// Key-value db internal failure
     StorageInternalError,
     /// Requested trie value by its hash which is missing in storage.
-    /// TODO (#8997): consider including hash of trie node.
-    MissingTrieValue,
+    MissingTrieValue(String),
```

Comment:
Nice idea to add more context in the message. 
I would make it more structrured by having the context and the hash separately
e.g.
```
MissingTrieValue(MissingTrieValueContext, CryptoHash),
```
Where the MissingTrieValueContext is an enum. The naming is up to you, what ever you think will fit here best. I would move the errors messages as you have them now to comments on the relevant enum values. 

This way it's typed and structured so that we can separately check the context and the hash if needed (e.g. in tests).

Reply:
Amazing idea! Will fix